### PR TITLE
Ensure overlay.Driver.recreateSymlinks does not loop endlessly

### PR DESCRIPTION
**Pull request by mtrmac**
- Add missing error handling so that if one `Remove` keeps failing, we don’t loop forever
- As a failsafe (if the filesystem were somehow completely broken), terminate the loop after 10 iterations. (There probably shouldn’t even _need_ to be a loop, but that would be a more invasive and risky change.)

See the individual commit messages for details.

WARNING: Completely untested.

### CODE REVIEW

**Comment by mtrmac** [comment]
(Overall I find the purpose of `recreateSymlinks` rather unclear. Why do we expect _these particular two files_ (`overlay/l/…` and `overlay/…/link`) to be corrupt, but not any other critical files that make up the layer storage?!)

**Review by giuseppe — APPROVED** [review]
LGTM

File: drivers/overlay/overlay.go [hunk 1]
```diff
@@ -1233,7 +1238,10 @@ func (d *Driver) recreateSymlinks() error {
 			if len(targetComponents) != 3 || targetComponents[0] != ".." || targetComponents[2] != "diff" {
 				errs = multierror.Append(errs, errors.Errorf("link target of %q looks weird: %q", link, target))
 				// force the link to be recreated on the next pass
-				os.Remove(filepath.Join(linksDir, link.Name()))
+				if err := os.Remove(filepath.Join(linksDir, link.Name())); err != nil {
```
Comment (rhatdan) [inline drivers/overlay/overlay.go:1241]: Should this error if the error is ENOEXIST?

Comment (mtrmac) [reply drivers/overlay/overlay.go:1241]: Good point — that shouldn’t be reported as an error, but it also shouldn’t be marked as having made progress.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**